packages/fxa-shared/test/metrics/amplitude.js

```diff
@@ -488,21 +488,36 @@ describe('metrics/amplitude:', () => {
       assert.isTrue(result);
     });
 
-    it('success - if event_type is `fxa_pay_setup - *` and language is provided', () => {
+    it('success - if event_type is `fxa_pay_setup - *` and language/product/plan/checkoutType is provided', () => {
```

Comment:
I think this is getting complex enough here that we can add some nested `describe` blocks to make this easier to follow?
 * `describe` block for payments events global properties
   * `it` block for required properties (`language`)
   * `it` block for optional properties (`country_code`)
 * `describe` block for `fxa_pay_setup - *`
   * `it` block for required properties
   * `it` block for optional properties
 * `describe` block for `fxa_pay_subscription_change - *`
   * `it` block for required properties
   * `it` block for optional properties

Reply:
Blocking change request:

Still looking for the nested describe blocks here, so I can see what's covered/not covered still more readily.

In general, Lisa's [PR](https://github.com/mozilla/fxa/pull/15238) for the new `fxa_subscribe - subscription_ended` event is a good place to look for the kinds of test cases I'm looking for.

For example, I don't see a test case for `- fail` events checking for the required `error_id` property or a test case for `- success` events requiring the `country_code_source` property.

Reply:
tests updated